experiments/tf_trainer/common/text_preprocessor.py

```diff
@@ -32,13 +30,57 @@ class TextPreprocessor():
   """
 
   def __init__(self, embeddings_path: str) -> None:
-    self._word_to_idx, self._embeddings_matrix, self._unknown_token = TextPreprocessor._get_word_idx_and_embeddings(
-        embeddings_path)  # type: Tuple[Dict[str, int], np.ndarray, int]
+    self._word_to_idx, self._embeddings_matrix, self._unknown_token = \
+        TextPreprocessor._get_word_idx_and_embeddings(
+            embeddings_path)  # type: Tuple[Dict[str, int], np.ndarray, int]
+
+  def tokenize_tensor_op_tf_func(self) -> Callable[[types.Tensor], types.Tensor]:
+    """Tensor op that converts some text into an array of ints that correspond
+    with this preprocessor's embedding.
+
+    This function is implemented only with TensorFlow operations and is
+    therefore compatible with TF-Serving (vs. tokenize_tensor_op_py_func).
+    """
 
-  def tokenize_tensor_op(self, tokenizer: Callable[[str], List[str]]
-                        ) -> Callable[[types.Tensor], types.Tensor]:
+    vocabulary_table = tf.contrib.lookup.HashTable(
+        tf.contrib.lookup.KeyValueTensorInitializer(
+            keys=list(self._word_to_idx.keys()),
+            values=list(self._word_to_idx.values()),
+            key_dtype=tf.string,
+            value_dtype=tf.int64),
+        default_value=self._unknown_token)
+
+    def _tokenize_tensor_op(text: types.Tensor) -> types.Tensor:
+      '''Converts a string Tensor to an array of integers.
+
+      Args:
+        text: must be a scalar string tensor (rank 0).
+
+      Returns:
+        A 1-D Tensor of word integers.
+      '''
+
+      # TODO: Improve tokenizer.
+      # TODO: Add utf-8 encoding.
```

Comment:
Could you expand on this TODO a bit? How does this function currently handle utf-8 text?

Reply:
Added some info.